### PR TITLE
Allow using external fmt

### PR DESCRIPTION
**Pull request by adayton1**

### CODE REVIEW

File: src/umpire/event/CMakeLists.txt [hunk 1]
```diff
@@ -17,7 +17,7 @@ set (umpire_event_sources
   json_file_store.cpp
   recorder_factory.cpp)
 
-set (umpire_event_depends umpire_tpl_fmt umpire_tpl_json camp)
+set (umpire_event_depends fmt::fmt umpire_tpl_json camp)
```
Comment (adayton1) [inline src/umpire/event/CMakeLists.txt:20]: Do we want to use the header only library?

File: src/umpire/util/error.hpp [hunk 2]
```diff
@@ -10,7 +10,7 @@
 #include <stdexcept>
 #include <string>
 
-#include "umpire/fmt/fmt.hpp"
+#include "fmt/format.h"
```
Comment (adayton1) [inline src/umpire/util/error.hpp:13]: Are we OK with exposing fmt in a header?

**Comment by adayton1** [comment]
@davidbeckingsale, what should I do about the spack package for Umpire needing fmt? There's a circular dependency between radiuss-spack-packages and Umpire.

**Comment by adayton1** [comment]
I know I need to add something like:

`depends_on("fmt@9.1:10", when="@develop")`

File: src/umpire/util/error.hpp (hunk 2 above)
Comment (adayton1) [reply src/umpire/util/error.hpp:13]: Looks like it's not even used in this file. So now it shouldn't bleed out of Umpire's headers.

Comment (adayton1) [reply src/umpire/util/error.hpp:13]: Oh, but it will propagate into other headers, now.

File: src/umpire/event/CMakeLists.txt (hunk 1 above)
Comment (adayton1) [reply src/umpire/event/CMakeLists.txt:20]: The library is built, so I won't worry about it for now.

File: src/umpire/util/error.hpp (hunk 2 above)
Comment (adayton1) [reply src/umpire/util/error.hpp:13]: It would be nice to avoid this, but I think that's better left as a future exercise.

**Comment by adayton1** [comment]
It passed!

**Review by kab163 — APPROVED** [review]
Looks good. I no longer see all the fmt notes/warnings when building this branch with the same cmake command as the one from my issue!

**Pull request merged**